### DISCUSSION TITLE
Need help with Inverse abel transformation of an image collected using Velocity map imaging

### DISCUSSION THREAD

**Discussion by JBN55** [PyAbel/PyAbel#377] [Q&A] [answered]
Hi,

I am trying to do an inverse abel transform on a image I collected using a velocity map imaging setup. I am new to this whole technique and have managed to do the transform after going through some of the examples in this repository. I am attaching the original image and the inverse abel transform below, what I dont understand is the vertical  line in the transformed image. Can anyone help me understand this? 
![img_50](https://github.com/PyAbel/PyAbel/assets/72686611/e52d1837-f19a-467d-a3ea-70782f8e32c0)
![img_50_output](https://github.com/PyAbel/PyAbel/assets/72686611/8946fee4-c9af-4151-8be9-d9be86b4cc9e)

Thanks,
Jibin

**Comment by MikhailRyazanov** [comment, marked as answer]
1. Make sure to center the image properly before transforming it, either by [`abel.tools.center.center_image`](https://pyabel.readthedocs.io/en/latest/abel.html#abel.tools.center.center_image) or using `origin=` in [`abel.Tranform`](https://pyabel.readthedocs.io/en/latest/abel.html#abel.transform.Transform).
2. The vertical line is usually called “centerline noise” and is unavoidable in the inverse Abel transform. Basically, the amount of signal in the original 3D distribution is directly proportional to the distance from the axis (due to the _ρ_ _dρ_ in [cylindrical coordinates](https://en.wikipedia.org/wiki/Cylindrical_coordinate_system)), but the noise is its 2D projection is distributed more uniformly (in fact, the noise from larger radii overlaps with the signal at smaller radii, so the axial part suffers the most), thus the SNR in the inverse transform drops significantly towards the axis. You can try using a method with regularization to improve the situation.
3. If this makes sense for your data, try “polar” methods (`rbasex` or `linbasex`). They don't have this problem (although the very center of the transformed image will still be noisy).
4. If you're only interested in the speed (radial) distribution, then all this centerline noise doesn't really matter because sin θ dθ ≈ 0 near the axis (θ close to 0° and 180°).

**Reply by stggh** [reply]
Here is some example code which may help:
```python
import abel
import matplotlib.pyplot as plt

im = plt.imread('img.png')
print(f'image shape {im.shape}')

# PyAbel analysis -----------------------
ar_im = abel.Transform(im, method='rbasex', origin='com')
ah_im = abel.Transform(im, method='hansenlaw', origin='com',
                       angular_integration=True)

# plot ----------------------------------
fig, (ar, ah) = plt.subplots(1, 2)

ar.plot(*ar_im.distr.rIbeta()[:2])
ar.set_xlabel('radius (pixels)')
ar.set_title('rbasex Law')
ar.set_xlabel('radius (pixels)')
ar.set_ylabel('intensity')

ah.plot(*ah_im.angular_integration)
ah.set_title('Hansen Law')
ah.set_xlabel('radius (pixels)')

plt.savefig('vmi.svg')
plt.show()
```
![vmi](https://github.com/PyAbel/PyAbel/assets/10932229/3e84cfe7-68ec-40cd-8680-b7205aae5f5f)


Notes:

1. `img.png` is copied from the monochrome image that you gave above. 
2. The center origin needs to be determined. `origin='com'` estimates this based on the center of mass.

**Reply by JBN55** [reply]
Thank you very much for the help.
